ports/libsimpleservo/capi/src/lib.rs

```diff
@@ -167,10 +183,17 @@ pub extern "C" fn init_with_gl(
     opts: CInitOptions,
     wakeup: extern "C" fn(),
     callbacks: CHostCallbacks,
-) {
-    init_logger();
-    let gl = gl_glue::gl::init().unwrap();
-    unsafe { init(opts, gl, None, None, wakeup, callbacks) }
+) -> bool {
+    let _result = panic::catch_unwind(|| {
```

Comment:
Given how frequently we repeat this pattern of catch_unwind followed by checking its return value, we can make a helper to reduce some duplication:
```rust
fn catch_any_panic<F: FnOnce() + UnwindSafe>(function: F) -> bool {
    catch_unwind(function).is_ok()
}
```
Then we can write the calling code like:
```rust
extern "C" fn perform_updates() -> bool {
    catch_any_panic(|| {
        debug!("perform_updates");
        call(|s| s.perform_updates());
    })
}
```
This will make it easier to maintain in the future if we want to add additional behaviour for logging information about panics, for example.

Reply:
I modeled this idea after [this code](https://github.com/servo/rust-mozjs/blob/36d53b8d62fe70dbfc52da1fb9b828da30fdf33c/src/panic.rs#L18) we wrote a few years ago for a similar purpose.

Reply:
OK.  I will make the changes.  I was under the impression/misunderstanding on .is_ok() will return true when there is no panic, but was wondering how you return false if no explicit statement to do so.

Reply:
Result::is_ok() returns true if the value is Ok, and false if the value is Err. panic::catch_unwind returns Result<???> where ??? is the return type of the closure that it invokes, so Ok if there is no panic and Err otherwise. Therefore, catch_any_panic returns true if no panic occurred inside the closure passed to panic::catch_unwind.

Reply:
Can the generic function "catch_any_panic()" further be generalized to return a type based on the function signature?

I typed this up, doesn't build, but feel I'm close.

fn catch_any_panic<F: FnOnce() + UnwindSafe>(function: F) -> Result<T, E> {
    let result = panic::catch_unwind(function);

    match result {
        OK(T) => T,
        Err(E) => E,
    }
}

Reply:
It could (that's what [this code](https://github.com/servo/rust-mozjs/blob/36d53b8d62fe70dbfc52da1fb9b828da30fdf33c/src/panic.rs#L18) does), but then we end up having all of the callers that need a boolean return value have to do extra processing. It's not clear to me what the value of this change would be.

Reply:
I was thinking about the function pub extern "C" fn servo_version() -> *const c_char {} which returns a *const c_char.  I was trying to fit the catch_any_panic() to servo_version() too.

Reply:
I've got all the suggestions incorporated into my repository.  I will run a few tests, add, commit and push.

My question can be deferred.